mpqp/local_storage/load.py

```diff
@@ -0,0 +1,312 @@
+from __future__ import annotations
+
```

Comment:
what about a `get_result_by_remote_job_id` ?
because in my opinion the local job id should mostly be inner workings, it
doesn't need to be communicated to the user.
Maybe we still need it for local runs though